sdk/push/src/main/java/com/klaviyo/push/KlaviyoPushService.kt

```diff
@@ -26,26 +50,82 @@ class KlaviyoPushService : FirebaseMessagingService() {
         fun getCurrentPushToken(): String {
             return KlaviyoPreferenceUtils.readStringPreference(PUSH_TOKEN_PREFERENCE_KEY) ?: ""
         }
+
+        /**
+         * Logs an $opened_push event for a remote notification that originated from Klaviyo
+         *
+         * TODO customer and event properties should both be optional, once SDK is tracking customer state internally
+         *
+         * @param pushPayload The data attributes of the push notification payload
+         * @param customerProperties Profile with which to associate the event
+         * @param eventProperties Optional additional properties for the event
+         */
+        fun handlePush(
+            pushPayload: Map<String, String>,
+            customerProperties: KlaviyoCustomerProperties,
+            eventProperties: KlaviyoEventProperties? = null
+        ) {
+            if (pushPayload["origin"] != "Klaviyo") {
+                return
+            }
```

Comment:
is our backend already sending this in the payload?

Reply:
It will be added by [this app PR](https://github.com/klaviyo/app/pull/43923) which includes adding/correcting the tracking data in the android push payload.

Reply:
Changed backend to send `_k: json_string` for the tracking params, and i'll use that to identify push originated from us.